packages/jest-matchers/src/matchers.js

```diff
@@ -124,7 +124,10 @@ const matchers: MatchersObject = {
   },
 
   toMatch(actual: any, expected: string) {
-    if (!(expected instanceof RegExp) && !(typeof expected == 'string')) {
+    if (
+      !(typeof expected === 'string') &&
+      !(Object.prototype.toString.call(expected) === '[object RegExp]')
+    ) {
```

Comment:
@cpojer do you have any idea why the first variant might fail for some cases? In dev tools for the case
```js
  expect(diff('a', 'a')).toMatch(/no visual difference/);
```
 `expected.constructor` prints `[Function: RegExp]` but `expected.constructor === RegExp` is false.